internal/ringbuffer.go

```diff
@@ -0,0 +1,119 @@
+package internal
+
+import (
+	"fmt"
+	"github.com/hazelcast/hazelcast-go-client/core"
+	"github.com/hazelcast/hazelcast-go-client/internal/common/collection"
+	. "github.com/hazelcast/hazelcast-go-client/internal/protocol"
+)
+
+type RingbufferProxy struct {
+	*partitionSpecificProxy
+}
+
+func newRingbufferProxy(client *HazelcastClient, serviceName *string, name *string) (*RingbufferProxy, error) {
+	parSpecProxy, err := newPartitionSpecificProxy(client, serviceName, name)
+	if err != nil {
+		return nil, err
+	}
+	return &RingbufferProxy{parSpecProxy}, nil
+}
+
+func (rp *RingbufferProxy) Capacity() (capacity int64, err error) {
+	request := RingbufferCapacityEncodeRequest(rp.name)
+	responseMessage, err := rp.Invoke(request)
+	return rp.DecodeToInt64AndError(responseMessage, err, RingbufferCapacityDecodeResponse)
+}
+func (rp *RingbufferProxy) Size() (size int64, err error) {
```

Comment:
It looks like in map codecs `Size()` returns `int32` however in this one it is `int64`. In java it is also like that, is there a reason for that ?

Reply:
Yes you are right there is a inconsistency but it is out of scope of this PR.